sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousExecution.scala

```diff
@@ -60,6 +60,10 @@ class ContinuousExecution(
   // For use only in test harnesses.
   private[sql] var currentEpochCoordinatorId: String = _
 
+  // Throwable that caused the execution to fail
+  private var failure: Option[Throwable] = None
```

Comment:
Couldn't you just use an `AtomicReference` here instead of Option + lock?

Reply:
Checking...

Reply:
Changed.